### DISCUSSION TITLE
Pressure seems to countains only kinetic contribution

### DISCUSSION THREAD

**Discussion by Mathieu-Istas** [mir-group/pair_nequip#39] [General]
Hello,
Thanks for developing Nequip.
Possibly related to [this question](https://github.com/mir-group/nequip/discussions/309), but a bit different. I trained a model  that gives good predictions on forces and energies. However the value of the pressure is completely underestimated when I try to run MD simulations with Lammps, in the NPT or NVT ensemble with periodic boundary conditions. It is extremely close the ideal gas contribution (N k_b T / V), and the other contributions from the forces and periodic boundary conditions seems to be missing. What could be the cause?
It may be unrelated, but I also run into a segmentation fault if I increase the number of atoms in the system (even with 500 atoms is problematic). I wonder if it has to do with the boundary conditions.

**Comment by Linux-cpp-lisp** [comment]
Hi @Mathieu-Istas ,

Thanks for your interest in our code!

For the pressure, yes, getting this right requires the `stress` branch of `pair_nequip` at present.

For the segmentation fault, unfortunately this is sometimes how it fails when it runs out of memory, rather than a more useful error. This can be resolved generally with the use of a smaller model, a smaller cutoff, or the use of Allegro instead of NequIP to parallelize larger systems across multiple GPUs.

**Reply by Mathieu-Istas** [reply]
Thanks for the reply. I thought from the previous answer that I had to go on the branch `stress` of `nequip`. The branch `stress` of `nequip` is made to fit the stress tensor, is that correct?

**Reply by Linux-cpp-lisp** [reply]
Yes, sorry this is still pretty confusing as I manage the  merge-down.  The right combination is `develop` of `nequip` (into which `stress` was merged a while back) and `stress` of `pair_nequip`.

**Reply by Mathieu-Istas** [reply]
Thanks for the help, now I have an issue when running the command `make -j$(nproc)` to build lammps:
```
/home/molecular_dynamics/lammps/src/pair_nequip.cpp: In member function ‘virtual void LAMMPS_NS::PairNEQUIP::init_style()’:
/home/molecular_dynamics/lammps/src/pair_nequip.cpp:97:33: error: ‘int LAMMPS_NS::NeighRequest::half’ is protected within this context
   97 |   neighbor->requests[irequest]->half = 0;
 ```
I did try to clean everything before doing the make and I still have this issue. I am on the `stress` branch of `pair_nequip`. Here is the precise sha1 if that can help 97f3f7b20c0. Should I checkout somewhere else?

**Reply by Linux-cpp-lisp** [reply]
See, for example, https://github.com/mir-group/pair_nequip/issues/33

**Comment by Mathieu-Istas** [comment]
@Linux-cpp-lisp Thanks for the quick reply and the answer.s I had some technical difficulties to have the correct pytorch version, but I now have simulations running at the correct pressure.

**Reply by Linux-cpp-lisp** [reply]
Glad to hear!
